src/utils/encoding/cid_de_cbor.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use cid::serde::BytesToCidVisitor;
+use cid::Cid;
+use core::fmt;
+use serde::de::{self, DeserializeSeed, SeqAccess, Visitor};
+use serde::Deserializer;
+
+/// [`CidVec`] allows for efficient zero-copy de-serialization of `DAG_CBOR`-encoded nodes into a
+/// vector of [`Cid`].
+#[derive(Default)]
+pub struct CidVec(Vec<Cid>);
```

Comment:
I don't really like this name -it's too general - it was the original name for what is now `FrozenCids`, for example.

Can you think of a better name, or maybe different API? `pub fn filter_cids(_: ???)`

Reply:
Yeah, it's a good point. My idea behind this name is exactly that - this represents a `Vec<Cid>`. If we had a better way of dealing with ser/de without a wrapper - this wouldn't be needed. 

Perhaps a good middleground is to do something like: `pub fn extract_cids(cbor_blob: &bytes): Vec<Cid>`.

Reply:
It's done.